daemon/algod/api/spec/v1/model.go

```diff
@@ -542,6 +691,74 @@ type AssetFreezeTransactionType struct {
 	NewFreezeStatus bool `json:"freeze"`
 }
 
+// ApplicationCallTransactionType contains the additional fields for an ApplicationCall transaction
+// swagger:model ApplicationCallTransactionType
+type ApplicationCallTransactionType struct {
```

Comment:
All of these fields are marked as required, but that isn't really true is it? Could we make an OnCompletion / field matrix to define when they are required?

Reply:
Something similar to the Assets documentation on the developer site would be even better:

https://developer.algorand.org/docs/reference/transactions/#asset-transfer-transaction